tests/sizers/wrapsizer.cpp

```diff
@@ -24,104 +24,96 @@
-    const wxSize sizeChild2 = wxSize(sizeTotal.x/2 + 10, sizeTotal.y/3);
+    const wxSize sizeChild2 = wxSize(100, 80);
     sizeMinExpected.x += sizeChild2.x;
     sizeMinExpected.y = wxMax(sizeChild1.y, sizeChild2.y);
 
     wxWindow * const
-        child2 = new wxWindow(m_win, wxID_ANY, wxDefaultPosition, sizeChild2);
+        child2 = new wxWindow(win, wxID_ANY, wxDefaultPosition, sizeChild2);
     child2->SetBackgroundColour(*wxYELLOW);
-    m_sizer->Add(child2);
-    m_win->Layout();
+    sizer->Add(child2);
+    win->Layout();
 
-    CPPUNIT_ASSERT_EQUAL( sizeMinExpected, m_sizer->CalcMin() );
+    CHECK( sizeMinExpected == sizer->CalcMin() );
 
     // Three children will take at least two rows so the minimal size in
     // vertical direction must increase.
-    const wxSize sizeChild3 = wxSize(sizeTotal.x/2, sizeTotal.y/5);
+    const wxSize sizeChild3 = wxSize(90, 40);
     sizeMinExpected.y += sizeChild3.y;
 
     wxWindow * const
-        child3 = new wxWindow(m_win, wxID_ANY, wxDefaultPosition, sizeChild3);
+        child3 = new wxWindow(win, wxID_ANY, wxDefaultPosition, sizeChild3);
     child3->SetBackgroundColour(*wxGREEN);
-    m_sizer->Add(child3);
-    m_win->Layout();
+    sizer->Add(child3);
+    win->Layout();
+
+    CHECK( sizeMinExpected == sizer->CalcMin() );
+
+    delete win;
+}
+
+TEST_CASE("wxWrapSizer::CalcMinFromMinor", "[wxWrapSizer]")
+{
+    wxWindow *win = new wxWindow(wxTheApp->GetTopWindow(), wxID_ANY);
+    win->SetClientSize(180, 240);
+
+    wxSizer* boxSizer = new wxBoxSizer(wxHORIZONTAL);
+    win->SetSizer(boxSizer);
+
+    // To test CalcMinFromMinor function the wrap sizer with the
+    // horizonral align added to the box sizer with horizontal align.
+    wxSizer* wrapSizer = new wxWrapSizer(wxHORIZONTAL);
+    boxSizer->Add(wrapSizer);
+
+    // Add three child windows. Sum of the first and the second windows widths should
+    // be less than the width of the third window.
+    const wxSize sizeChild1 = wxSize(40, 60);
+    wxWindow * const
+        child1 = new wxWindow(win, wxID_ANY, wxDefaultPosition, sizeChild1);
+    child1->SetBackgroundColour(*wxRED);
+    wrapSizer->Add(child1);
+
+    const wxSize sizeChild2 = wxSize(50, 80);
+    wxWindow * const
+        child2 = new wxWindow(win, wxID_ANY, wxDefaultPosition, sizeChild2);
+    child2->SetBackgroundColour(*wxGREEN);
+    wrapSizer->Add(child2);
+
+    const wxSize sizeChild3 = wxSize(100, 120);
+    wxWindow * const
+        child3 = new wxWindow(win, wxID_ANY, wxDefaultPosition, sizeChild3);
+    child3->SetBackgroundColour(*wxBLUE);
+    wrapSizer->Add(child3);
+
+    // First two windows should be in a first row and the third in a second row.
+    const wxSize sizeMinExpected = wxSize(sizeChild3.x, sizeChild2.y + sizeChild3.y);
+    win->Layout();
+    CHECK(sizeMinExpected == wrapSizer->CalcMin());
 
-    CPPUNIT_ASSERT_EQUAL( sizeMinExpected, m_sizer->CalcMin() );
+    delete win;
```

Comment:
This should be handled using `wxScopedPtr` as windows remaining after the test end if something goes wrong in it screw up the subsequent tests which can be very annoying as it's usually not at all obvious that a failure in some completely unrelated test is due to this test failing and leaving its window on screen.